backend/api/src/model/block.rs

```diff
@@ -0,0 +1,176 @@
+//! Blocks that make up the content of realm pages.
+
+use anyhow::anyhow;
+use futures::TryStreamExt;
+use juniper::{graphql_interface, FieldResult, GraphQLEnum, GraphQLObject};
+use postgres_derive::FromSql;
+use postgres_types::{FromSql, ToSql};
+
+use tobira_util::prelude::*;
+use tokio_postgres::Row;
+use crate::{Context, Id, id::Key, util::RowExt};
+
+
+/// A `Block`: an UI element that belongs to a realm.
+#[graphql_interface(for = [Text, VideoList])]
+pub(crate) trait Block {
+    // To avoid code duplication, all the shared data is stored in `SharedData`
+    // and only a `shared` method is mandatory. All other method (in particular,
+    // all that are visible to GraphQL) are defined in the trait already.
+    #[graphql(skip)]
+    fn shared(&self) -> &SharedData;
+
+    fn id(&self) -> Id {
+        self.shared().id
+    }
+    fn realm_id(&self) -> Id {
+        self.shared().realm_id
+    }
```

Comment:
I think it would be more GraphQL-idiomatic to return a `Realm`, here, and get the ID from there if you need it.

Reply:
Ohhhh true.